Segfault in prepare job during Patching of the CoreDNS Configmap

When trying to install maesh on our Cluster, the prepare Job crashes with this segfault:

```
time="2019-10-02T08:48:11Z" level=info msg="Checking Cluster..."
time="2019-10-02T08:48:11Z" level=info msg="Cluster check Complete..."
time="2019-10-02T08:48:11Z" level=info msg="Preparing Cluster..."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x100c7d7]

goroutine 1 [running]:
github.com/containous/maesh/internal/k8s.(*ClientWrapper).patchCoreConfigMap(0xc00039a280, 0xc000104480, 0xc000447b60, 0x1, 0x1)
        /go/src/github.com/containous/maesh/internal/k8s/client.go:228 +0x67
github.com/containous/maesh/internal/k8s.(*ClientWrapper).patchCoreDNS(0xc00039a280, 0x13fde0c, 0x7, 0x1400ff0, 0xb, 0xc0004bde80, 0x14)
        /go/src/github.com/containous/maesh/internal/k8s/client.go:207 +0x14b
github.com/containous/maesh/internal/k8s.(*ClientWrapper).InitCluster(0xc00039a280, 0x13fcecf, 0x5, 0x0, 0xc00039a280)
        /go/src/github.com/containous/maesh/internal/k8s/client.go:191 +0x101
github.com/containous/maesh/cmd/prepare.patchCommand(0xc0002668c0, 0x20, 0x30)
        /go/src/github.com/containous/maesh/cmd/prepare/prepare.go:46 +0x35d
github.com/containous/maesh/cmd/prepare.NewCmd.func1(0xc0000b2030, 0x0, 0x0, 0x0, 0xc000283d50)
        /go/src/github.com/containous/maesh/cmd/prepare/prepare.go:20 +0x2a
github.com/containous/traefik/v2/pkg/cli.run(0xc000283d50, 0xc0000b2030, 0x0, 0x0, 0x8, 0xc0000d9d18)
        /go/pkg/mod/github.com/containous/traefik/v2@v2.0.0/pkg/cli/commands.go:117 +0x171
github.com/containous/traefik/v2/pkg/cli.execute(0xc000283d50, 0xc0000b2030, 0x1, 0x1, 0x0, 0xc000292201, 0xc00009f5c0)
        /go/pkg/mod/github.com/containous/traefik/v2@v2.0.0/pkg/cli/commands.go:46 +0x7ba
github.com/containous/traefik/v2/pkg/cli.execute(0xc000283ce0, 0xc0000b2020, 0x2, 0x2, 0xc000283d01, 0x1e8b920, 0xc000058750)
        /go/pkg/mod/github.com/containous/traefik/v2@v2.0.0/pkg/cli/commands.go:81 +0x15c
github.com/containous/traefik/v2/pkg/cli.Execute(...)
        /go/pkg/mod/github.com/containous/traefik/v2@v2.0.0/pkg/cli/commands.go:40
main.main()
        /go/src/github.com/containous/maesh/cmd/maesh/maesh.go:43 +0x3e1
```

The Cluster is a kubeadm cluster on version 1.14.7 with CoreDNS 1.3.1. Tiller is running with a `cluster-admin` ClusterRole so it should have enough permissions.

--- a/internal/k8s/client.go
+++ b/internal/k8s/client.go
@@ -331,7 +331,7 @@ func (w *ClientWrapper) patchCoreDNSConfigMap(coreDeployment *appsv1.Deployment)
 		return false, errors.New("coreDNS configmap not defined")
 	}
 
-	coreConfigMapName = coreDeployment.Spec.Template.Spec.Volumes[0].ConfigMap.Name
+	coreConfigMapName = getCoreDNSConfigMapName(coreDeployment)
 
 	coreConfigMap, err := w.KubeClient.CoreV1().ConfigMaps(coreDeployment.Namespace).Get(coreConfigMapName, metav1.GetOptions{})
 	if err != nil {
@@ -474,7 +474,7 @@ func (w *ClientWrapper) isCoreDNSPatched(deploymentName string, namespace string
 		return errors.New("coreDNS configmap not defined")
 	}
 
-	coreConfigMapName := coreDeployment.Spec.Template.Spec.Volumes[0].ConfigMap.Name
+	coreConfigMapName := getCoreDNSConfigMapName(coreDeployment)
 
 	coreConfigMap, err := w.KubeClient.CoreV1().ConfigMaps(coreDeployment.Namespace).Get(coreConfigMapName, metav1.GetOptions{})
 	if err != nil {
@@ -532,6 +532,16 @@ func buildSmiSplitClient(config *rest.Config) (*smiSplitClientset.Clientset, err
 	}
 
 	return client, nil
+}
+
+// getCoreDNSConfigMapName returns the dected coredns configmap name
+func getCoreDNSConfigMapName(coreDeployment *appsv1.Deployment) string {
+	for _, volume := range coreDeployment.Spec.Template.Spec.Volumes {
+		if volume.ConfigMap != nil {
+			return volume.ConfigMap.Name
+		}
+	}
+	return ""
 }
 
 // GetService retrieves the service from the specified namespace.

